src/store/Workspaces/index.ts

```diff
@@ -182,6 +183,21 @@ export const actionCreators: ActionCreators = {
     }
   },
 
+  restartWorkspace: (workspace: Workspace): AppThunk<KnownAction, Promise<void>> => async (dispatch, getState): Promise<void> => {
+    try {
+      const state = getState();
+      const cheDevworkspaceEnabled = state.workspacesSettings.settings['che.devworkspaces.enabled'] === 'true';
+      if (cheDevworkspaceEnabled && isWorkspaceV2(workspace.ref)) {
+        await dispatch(DevWorkspacesStore.actionCreators.restartWorkspace(workspace.ref));
+      } else {
+        await dispatch(CheWorkspacesStore.actionCreators.restartWorkspace(workspace.ref as che.Workspace));
+      }
+    } catch (e) {
+      dispatch({ type: 'RECEIVE_ERROR' });
+      throw new Error(`Failed to restart the workspace, ID: ${workspace.id}, ` + e.message);
```

Comment:
To be consistent let `DevWorkspaceStore` and `CheWorkspaceStore` create and throw their own error messages.
```suggestion
      throw e;
```

Reply:
I have fixed it.